src/browsers/tob/shared/components/DotplotHeatmap.js

```diff
@@ -0,0 +1,241 @@
+          </linearGradient>
+        </defs>
+        <g transform={`translate(${margin.left}, ${margin.top})`}>
+          <g id="x-axis">
+            <line x2={`${innerWidth}`} stroke="black" />
+            <line y1={`${innerHeight}`} y2={`${innerHeight}`} x2={`${innerWidth}`} stroke="black" />
+            {xScaleLocal.domain().map((tick) => (
+              <g
+                key={tick}
+                transform={`translate(${
+                  xScaleLocal(tick) + xScaleLocal.bandwidth() / 2
+                }, ${innerHeight})`}
+              >
+                <text
+                  transform="translate(0, 10)rotate(-90)"
+                  style={{ textAnchor: 'end', alignmentBaseline: 'middle' }}
+                >
+                  {tick}
+                </text>
+                <line y2={6} stroke="black" />
+                <line y2={`-${innerHeight}`} stroke="lightgrey" />
+              </g>
+            ))}
+          </g>
+          <g id="y-axis">
+            <line y2={`${innerHeight}`} stroke="black" />
+            <line x1={`${innerWidth}`} y2={`${innerHeight}`} x2={`${innerWidth}`} stroke="black" />
+            {yScaleLocal.domain().map((tick) => (
+              <g
+                key={tick}
+                transform={`translate(0, ${yScaleLocal(tick) + yScaleLocal.bandwidth() / 2})`}
+              >
+                <text key={tick} style={{ textAnchor: 'end', alignmentBaseline: 'middle' }} x={-6}>
+                  {tick}
+                </text>
+                <line x2={-3} stroke="black" />
+                <line x2={`${innerWidth}`} stroke="lightgrey" />
+              </g>
+            ))}
+          </g>
+          <g id="tiles">
+            {data.length ? (
+              data.map((item) => (
+                <circle
+                  key={`${accessors.x(item)},${accessors.y(item)}`}
+                  r={sizeScaleLocal(accessors.size(item))}
+                  cx={xScaleLocal(accessors.x(item)) + xScaleLocal.bandwidth() / 2}
+                  cy={yScaleLocal(accessors.y(item)) + yScaleLocal.bandwidth() / 2}
+                  stroke="black"
+                  fill={`${colorScaleLocal(accessors.color(item))}`}
+                  onClick={onClick}
+                />
+              ))
+            ) : (
+              <g transform={`translate(${innerWidth / 2}, ${innerHeight / 2})`}>
+                <text
+                  style={{ textAnchor: 'middle', alignmentBaseline: 'middle', fontSize: 'larger' }}
+                >
+                  No data to display.
+                </text>
+              </g>
+            )}
+          </g>
+          {data.length && (
+            <g
+              id="sizeLegend"
+              transform={`translate(${
+                width - margin.right - margin.left + sizeScaleLocal.range()[1] + 5
+              }, 20)`}
+            >
+              <text fontWeight="bold"> P value </text> <br />
+              {sizeScaleLocal
+                .ticks(6)
```

Comment:
Reduce ticks to 3 in both the size and colour legends. That should be sufficient to convey the nature of the scale while eliminating visual clutter.

Reply:
Changed for now but note that scale.ticks(3) is only a suggestion. Sometimes it will do fewer or more ticks if 3 doesn't make nice numbers. To force it to use always 3 we'll have to do tickValues with the min, max and middle

Reply:
Yeah, I reading about this d3 behavior. It's fine for now - I think we'll end up moving legends to a separate component anyway at some point.